monetio/models/ufs.py

```diff
@@ -1053,69 +1053,76 @@ def dict_species_sums(mech):
         )
 
     else:
-        raise NotImplementedError(
-            "Mechanism not supported, update _rrfs_cmaq_mm.py file in MONETIO"
-        )
+        raise NotImplementedError("Mechanism not supported, update ufs.py file in MONETIO")
 
     return sum_dict
 
 
-def _calc_hgt(f):
-    """Calculates the geopotential height in m from the variables hgtsfc and
-    delz. Note: To use this function the delz value needs to go from surface
-    to top of atmosphere in vertical. Because we are adding the height of
-    each grid box these are really grid top values
+def _calc_hgt(dset: xr.Dataset) -> xr.DataArray:
+    """Calculate the geopotential height in m.
 
     Parameters
     ----------
-    f : xarray.Dataset
-        RRFS-CMAQ model data
+    dset : xarray.Dataset
+        The UFS dataset
 
     Returns
     -------
-    xr.DataArray
-        Geoptential height with attributes.
+    xarray.DataArray
+        Geopotential height
     """
-    sfc = f.surfalt_m.load()
-    dz = f.dz_m.load() * -1.0
-    # These are negative in RRFS-CMAQ, but you resorted and are adding from the surface,
-    # so make them positive.
-    dz[:, 0, :, :] = dz[:, 0, :, :] + sfc  # Add the surface altitude to the first model level only
-    z = dz.rolling(z=len(f.z), min_periods=1).sum()
+    # Get surface altitude
+    sfc = dset.surfalt_m
+
+    # Get the vertical displacement and flip sign as needed
+    dz = dset.dz_m * -1.0
```

Comment:
```suggestion
    dz = dset.dz_m
```
It's already been fixed before we call this func.

We should probably also move the fixing it (L209 below) inside the first cond branch.
https://github.com/noaa-oar-arl/monetio/blob/bbc0d8218f7128ca5d015b9bd8337fa7b116922a/monetio/models/ufs.py#L205-L209

Reply:
> # Get the vertical displacement and flip sign as needed
    dz = dset.dz_m * -1.0

Removing this line changed answers, and I'm not sure how it interacts with the suggested change about moving the in the main function. Moving the sign switch inside the conditional did not address the failure. I'm inclined to leave as is without additional testing, but let me know...

Reply:
The "as needed" is the key there. It's not needed, since the sign has already been flipped before we call this function (although that should only be done in the case that the first condition in that block is true, not always done like it is now, which is a bug, which we can fix here).

I tested this branch, and I could see the heights were negative, which is not what we want (regardless of what it was before). Might as well fix it in this PR. Suggest adding to your test some assertions related to the height and pressure (min/max at expected ends of the _z_ dim, all positive, monotonicity, etc.).

Reply:
Yeah, I noticed they were negative. I'll fix and update the baseline.

Reply:
I made the changes as suggested and updated the baseline. The new value description for `alt_msl_m_full`:
```
mean     15257.173828
std      13814.880859
min        785.901489
25%       3529.174255
50%      10933.270508
75%      23337.199707
max      60205.074219
Name: alt_msl_m_full, dtype: float64
```